src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -37,28 +39,77 @@ public class LoggingSource
 
         private static byte[] _headerRow =
             Encoding.UTF8.GetBytes("Time,\tThread,\tLevel,\tSource,\tLogger,\tMessage,\tException");
-        
-        private readonly ConcurrentDictionary<WebSocket, TaskCompletionSource<object>> _listeners = new ConcurrentDictionary<WebSocket, TaskCompletionSource<object>>();
+
+        public class WebSocketContext
+        {
+            public TaskCompletionSource<object> TaskCompletion { get; } = new TaskCompletionSource<object>();
+            public LogginFilter Filter { get; } = new LogginFilter();
+        }
+        private readonly ConcurrentDictionary<WebSocket, WebSocketContext> _listeners = new ConcurrentDictionary<WebSocket, WebSocketContext>();
+
         private LogMode _logMode;
         private LogMode _oldLogMode;
 
-        public async Task Register(WebSocket source)
+        public async Task<string> ReadFromWebsocket(WebSocket source, CancellationTokenSource sourceToken)
+        {
+            ArraySegment<byte> buffer = new ArraySegment<byte>(new byte[1024]);
```

Comment:
This allocates multiple objects on every call, we want to make sure that they are allocated once, per connection